src/main/java/org/opensearch/flowframework/transport/ProvisionWorkflowTransportAction.java

```diff
@@ -0,0 +1,166 @@
+     */
+    @Inject
+    public ProvisionWorkflowTransportAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        ThreadPool threadPool,
+        Client client,
+        WorkflowProcessSorter workflowProcessSorter
+    ) {
+        super(ProvisionWorkflowAction.NAME, transportService, actionFilters, WorkflowRequest::new);
+        this.threadPool = threadPool;
+        this.client = client;
+        this.workflowProcessSorter = workflowProcessSorter;
+    }
+
+    @Override
+    protected void doExecute(Task task, WorkflowRequest request, ActionListener<WorkflowResponse> listener) {
+
+        // Retrieve use case template from global context
+        String workflowId = request.getWorkflowId();
+        GetRequest getRequest = new GetRequest(GLOBAL_CONTEXT_INDEX, workflowId);
+
+        // Stash thread context to interact with system index
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.wrap(response -> {
+                context.restore();
+
+                // Parse template from document source
+                Template template = Template.parseFromDocumentSource(response.getSourceAsString());
+
+                // TODO : Update state index entry to PROVISIONING, given workflowId
+
+                // Respond to rest action then execute provisioning workflow async
+                listener.onResponse(new WorkflowResponse(workflowId));
+                executeWorkflowAsync(workflowId, template.workflows().get(PROVISION_WORKFLOW));
+            }, exception -> {
+                logger.error("Failed to retrieve template from global context.", exception);
+                listener.onFailure(new FlowFrameworkException(exception.getMessage(), RestStatus.INTERNAL_SERVER_ERROR));
+            }));
+        } catch (Exception e) {
+            logger.error("Failed to retrieve template from global context.", e);
+            listener.onFailure(new FlowFrameworkException(e.getMessage(), RestStatus.INTERNAL_SERVER_ERROR));
+        }
+    }
+
+    /**
+     * Retrieves a thread from the provision thread pool to execute a workflow
+     * @param workflowId The id of the workflow
+     * @param workflow The workflow to execute
+     */
+    private void executeWorkflowAsync(String workflowId, Workflow workflow) {
+        // TODO : Update Action listener type to State index Request
+        ActionListener<String> provisionWorkflowListener = ActionListener.wrap(response -> {
+            logger.info("Provisioning completed successuflly for workflow {}", workflowId);
+
+            // TODO : Create State index request to update STATE entry status to READY
+        }, exception -> {
+            logger.error("Provisioning failed for workflow {} : {}", workflowId, exception);
+
+            // TODO : Create State index request to update STATE entry status to FAILED
```

Comment:
is there an issue for this TODO?

Reply:
https://github.com/opensearch-project/opensearch-ai-flow-framework/issues/74